src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -560,6 +563,33 @@ public void startJenkins() throws Throwable {
             }
             Thread.sleep(100);
         }
+        addTimeout();
+    }
+
+    public static Optional<String> checkResult(HttpURLConnection conn) throws IOException {
+
+        int code = conn.getResponseCode();
+        System.out.println("Response Code "+ code);
+        if (code == 200) {
+            conn.getInputStream().close();
+            return Optional.empty();
+        } else {
+            String err = "?";
+            try (InputStream is = conn.getErrorStream()) {
+                if (is != null) {
+                    err = IOUtils.toString(is, StandardCharsets.UTF_8);
+                }
+            } catch (Exception x) {
+                x.printStackTrace();
+            }
+            if (code == 500) {
+                throw new JenkinsStartupException("Jenkins failed to start");
```

Comment:
keep the `err` message so that you can check the error if needed (and it is not lost after the fact)